src/Colopl/Spanner/Connection.php

```diff
@@ -443,4 +462,83 @@ protected function runQueryCallback($query, $bindings, Closure $callback)
 
         return $result;
     }
+
+    /**
+     * Returns current mode
+     *
+     * @return string
+     */
+    protected function getSessionNotFoundMode()
+    {
+        return $this->config['sessionNotFoundErrorMode'] ?? self::CLEAR_SESSION_POOL;
+    }
+
+    /**
+     * Handle "session not found" errors
+     *
+     * @template T
+     * @param  Closure(): T $callback
+     * @return T
+     * @throws InvalidArgumentException|NotFoundException|AbortedException
+     */
+    protected function sessionNotFoundWrapper(Closure $callback)
```

Comment:
Can you rename this to something like
```suggestion
    protected function withSessionHandling(Closure $callback): mixed
```

Reply:
fixed in recent commit